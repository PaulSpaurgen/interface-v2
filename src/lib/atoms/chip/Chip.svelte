<script lang="ts">
	import type { AppVariant } from '$lib/types/componentTypes';

	export let variant: AppVariant = 'primary';
	export let styles = '';
	const chipStyle =
		'rounded-3xl px-[20px] py-[9px] flex items-center gap-1.5 text-sm tracking-widest';

	const getClassFromVariant = (variant: AppVariant) => {
		switch (variant) {
			case 'primary':
				return 'bg-base-200 text-primary font-semibold';

			case 'secondary':
			default:
				return 'bg-grey-200 text-grey-500';
		}
	};
</script>

<div class={`${chipStyle} ${styles} ${getClassFromVariant(variant)}`}>
	<slot />
</div>
