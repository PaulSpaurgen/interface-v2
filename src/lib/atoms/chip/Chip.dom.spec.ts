import Chip from './Chip.svelte';
import { describe, test, afterEach } from 'vitest';
import { cleanup, render } from '@testing-library/svelte';

describe('Chip', () => {
    afterEach(() => {
        cleanup();
    });

    test('it should render the chip item', () => {
        const { getByTestId, container } = render(Chip);
        expect(() => getByTestId('chip-item')).not.toThrow();
        expect(container).toMatchSnapshot()
    });

    test('it should render the chip item with variant', () => {
        const { container: secondaryContainer } = render(Chip, { variant: 'secondary' });
        expect(secondaryContainer).toMatchSnapshot()
    });

    test('it should have styles passed as props its base classes attached to it', () => {
        const { getByTestId } = render(Chip, { props: { styles: 'bg-base-200' } });
        expect(getByTestId('chip-item').className).toContain('bg-base-200');
    });
});
