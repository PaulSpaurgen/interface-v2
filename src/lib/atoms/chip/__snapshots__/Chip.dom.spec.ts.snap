// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chip > renders the chip item 1`] = `
<body>
  <div>
    <div
      class="flex items-center gap-1.5 rounded-3xl px-[20px] py-[9px] text-sm tracking-widest  bg-base-200 text-primary font-semibold"
      data-testid="chip-item"
    />
    <!--&lt;Chip&gt;-->
  </div>
</body>
`;

exports[`Chip > renders the chip item with variant 1`] = `
<body>
  <div>
    <div
      class="flex items-center gap-1.5 rounded-3xl px-[20px] py-[9px] text-sm tracking-widest  bg-grey-200 text-[#6f6f71]"
      data-testid="chip-item"
    />
    <!--&lt;Chip&gt;-->
  </div>
</body>
`;
